tests/Unit/Core/Kpi/KpiRowPresenterTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * 2007-2019 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\Core\Kpi;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Kpi\KpiInterface;
+use PrestaShop\PrestaShop\Core\Kpi\Row\KpiRowInterface;
+use PrestaShop\PrestaShop\Core\Kpi\Row\KpiRowPresenter;
+use PrestaShop\PrestaShop\Core\Kpi\Row\KpiRowPresenterInterface;
+
+class KpiRowPresenterTest extends TestCase
+{
+    public function testCanBeConstructed()
+    {
+        $presenter = new KpiRowPresenter();
+
+        $this->assertInstanceOf(KpiRowPresenterInterface::class, $presenter);
```

Comment:
Isn't it obvious with `final class KpiRowPresenter implements KpiRowPresenterInterface` and no constructor which may throw an exception?

Reply:
It's a non-regression tests that does not check the current implementation but rather attempts to make sure than any evolution of the KpiRowPresenter class will keep implementing KpiRowPresenterInterface.

In some projets, I have seen 1 interface being implemented by 3 classes. Over time, the interface evolved, 2 of the classes were updated accordingly but the last class was left unmodified and this introduced bugs months later. This test attempts to prevent that.